app/handler/unsubscribe.py

```diff
@@ -0,0 +1,196 @@
+from email.message import Message
+
+from aiosmtpd.smtp import Envelope
+
+from app.config import URL
+from app.db import Session
+from app.email import headers, status
+from app.email_utils import send_email, render, add_or_replace_header
+from app.log import LOG
+from app.models import Alias, Contact, User, Mailbox
+
+
+class UnsubscribeGenerator:
+    def _generate_header_with_original_behaviour(self, message: Message) -> Message:
+        unsubscribe_data = message[headers.LIST_UNSUBSCRIBE]
+        if not unsubscribe_data:
+            return message
+
+    def _generate_header_with_sl_behaviour(
+        self, alias: Alias, contact: Contact, message: Message
+    ) -> Message:
+        user = alias.user
+        if user.one_click_unsubscribe_block_sender:
+            unsubscribe_link, via_email = alias.unsubscribe_link(contact)
+        else:
+            unsubscribe_link, via_email = alias.unsubscribe_link()
+
+        add_or_replace_header(
+            message, headers.LIST_UNSUBSCRIBE, f"<{unsubscribe_link}>"
+        )
+        if not via_email:
+            add_or_replace_header(
+                message, headers.LIST_UNSUBSCRIBE_POST, "List-Unsubscribe=One-Click"
+            )
+        return message
+
+    def add_header_to_message(
+        self, alias: Alias, contact: Contact, message: Message
+    ) -> Message:
+        """
+        Add List-Unsubscribe header
+        """
+        user = alias.user
+        if user.setting:
```

Comment:
is `user.setting` a new column?

Reply:
it's just a placeholder for the new setting. The setting itself is not yet there. Any ideas how to call it? I'm really bad at naming..